wolfssl/wolfcrypt/types.h

```diff
@@ -1808,6 +1808,9 @@ typedef struct w64wrapper {
         #define wc_static_assert(expr) struct wc_static_assert_dummy_struct
         #define wc_static_assert2(expr, msg) wc_static_assert(expr)
     #elif !defined(wc_static_assert)
+        #if defined(WOLFSSL_HAVE_ASSERT_H) && !defined(WOLFSSL_NO_ASSERT_H)
```

Comment:
I'm concerned about the current handling of [setting A] and [setting B] when neither is defined. It seems redundant with the NO_DEFINE. This pattern has started to creep into the library of late where historically we stuck to a "It is defined or default OFF" or "It is not defined or default ON".

I feel we should be establishing a clear default and remove the need for the NO_DEFINE state or vice-versa. I think that will improve the clarity of the code and lead to less confusion about which is the default behavior.

Your thoughts?

Reply:
`WOLFSSL_HAVE_ASSERT_H` is defined by the autotools configuration through autosensing, but we need to have a simple unambiguous way for users to override the autotools decision if necessary (e.g. in `user_settings.h` but also `CFLAGS=`).

Reply:
Got it. OK that makes sense so we're saying:

A) If autotools defined it && user does NOT wish to override go ahead and include the header
B) If autotools did NOT define it don't include the header
C) If autotools did define it && user DOES with to override, do not include.

Tracking. Appreciate the explanation!